### DISCUSSION TITLE
version error

### DISCUSSION THREAD

**Discussion by navika2025** [Project-MONAI/monai-deploy-app-sdk#473] [General]
@MMelQin 
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
typeguard 4.1.5 requires typing-extensions>=4.7.0; python_version < "3.12", but you have typing-extensions 4.5.0 which is incompatible.

when I installed a version greater than this it shows to download the previous version

**Comment by MMelQin** [comment]
The typeguard is not pinned, rather with a min version of 3.0.0 in the SDK, so the newest version will be installed when setting up virtual env and on build MAP container image.

In testing in Python 3.8 and 3.10 env, typeguard 4.2.1 is seen installed, but there is no typing-extension. So, the specific issue is not observed.

In the newly released App SDK v1.0, it is known that the pip is pinned at 23.3.2 while the latest is 24.0, and an error is shown. This is a known issue introduced by the underlying Holoscan v1.0.3 for it pins Python packages for long term support. This issue will be addressed in the next release.
